src/com/facebook/buck/go/GoLibraryDescription.java

```diff
@@ -57,21 +69,59 @@ public Arg createUnpopulatedConstructorArg() {
   }
 
   @Override
+  public boolean hasFlavors(ImmutableSet<Flavor> flavors) {
+    return goBuckConfig.getPlatformFlavorDomain().containsAnyOf(flavors);
+  }
+
+  @Override
+  public <A extends Arg, U> Optional<U> createMetadata(
+      BuildTarget buildTarget,
+      final BuildRuleResolver resolver,
+      A args,
+      Class<U> metadataClass) throws NoSuchBuildTargetException {
+    if (metadataClass.isAssignableFrom(Arg.class)) {
+      return Optional.of(metadataClass.cast(args));
```

Comment:
I love this, can you move it to https://github.com/facebook/buck/blob/master/src/com/facebook/buck/rules/BuildRuleResolver.java#L74 so we get this by default in all descriptions that implement `MetadataProvidingDescription`?

Actually let's go one step further: let's do it for all descriptions regardless of whether they implement `MetadataProvidingDescription` or not, as it's completely general.

Can you do it in a separate PR though?